mne/viz/_3d.py

```diff
@@ -464,6 +478,78 @@ def plot_source_estimates(stc, subject=None, surface='inflated', hemi='lh',
+        warnings.warn('Using fmin, fmid, fmax is deprecated and will be'
+                      ' removed in v0.10. Use "limits" instead.',
+                      DeprecationWarning)
+        for fi, f in enumerate(scale_pts):
+            if f is None:
+                scale_pts[fi] = [5., 10., 15.][fi]
+
+    # Using new cmap limit behavior
+    else:
+        if any([f is not None for f in [fmin, fmid, fmax]]):
+            warnings.warn('"limits" overrides fmin, fmid, fmax')
+
+        # Colormap is 'mne_analyze'. Need to construct cmap and get scale vals
+        if colormap == 'mne_analyze':
+            if limits == 'auto':
+                ctrl_pts = np.percentile(np.abs(stc.data), [97, 98.5, 99.9])
+                colormap = mne_analyze_colormap(ctrl_pts)
+            elif isinstance(limits, tuple):
+                assert len(limits) == 3, '"limits" tuple must be length 3'
+                ctrl_pts = limits
+                colormap = mne_analyze_colormap(ctrl_pts)
+            elif isinstance(limits, dict):
+                if limits['kind'] == 'percent':
+                    ctrl_pts = np.percentile(np.abs(stc.data),
+                                             list(np.abs(limits['pos_lims'])))
+                elif limits['kind'] == 'value':
+                    ctrl_pts = limits['pos_lims']
+                else:
+                    raise ValueError('limits[kind] must be "value" or '
+                                     ' "percent"')
+                if 'lims' in limits:
+                    warnings.warn('"limits[lim]" is not used if colormap is'
+                                  ' "mne_analyze." Use "pos_lims"')
+                colormap = mne_analyze_colormap(ctrl_pts)
+            else:
+                raise ValueError('"limits" must be "auto", tuple or dict')
+
+            scale_pts = [-1 * ctrl_pts[-1], 0, ctrl_pts[-1]]
+
+        # Standard colormap (not 'mne_analyze'). Only need to get scale vals
+        else:
+            if limits == 'auto':
+                scale_pts = np.percentile(np.abs(stc.data), [97, 98.5, 99.9])
+            elif isinstance(limits, tuple):
+                assert len(limits) == 3, '"limits" tuple must be length 3'
+                scale_pts = limits
+            elif isinstance(limits, dict):
+                if limits['kind'] == 'value':
+                    scale_pts = limits['lims']
+                elif limits['kind'] == 'percent':
+                    scale_pts = np.percentile(stc.data, list(limits['lims']))
+                else:
+                    raise ValueError('limits[kind] must be "value" or '
+                                     ' "percent"')
+
+                if 'pos_lims' in limits:
+                    warnings.warn('limits[pos_lims] not used if colormap is '
+                                  ' not mne_analyze. Use limits[lims]')
+            else:
+                raise ValueError('"limits" must be "auto", tuple or dict')
+
```

Comment:
this block of code looks repetitive. maybe we should have a private function that does the job and call it here with different parameters. Or am I missing something?